src/resources/dart-swift-concurrency.md

```diff
@@ -0,0 +1,303 @@
+In Dart, `ObservableObject` is simply represented 
+by either a `Future` or `Stream` object- 
+that represents values to be provided in the future.  
+So in this example you can return a `Future<Weather>` 
+from a function within the view model:
+
+<?code-excerpt "lib/async_weather.dart (HomePageViewModel)"?>
+```dart
+@immutable
+class HomePageViewModel {
+  const HomePageViewModel();
+  Future<Weather> load() async {
+    await Future.delayed(const Duration(seconds: 1));
+    return Weather.sunny;
+  }
+}
+```
+
+If you look at `load()` you can see 
+similarities to the Swift code. 
+The function is marked as `async` because internally, 
+it is using the `await` keyword.
+
+Additionally, inside a function marked as `async`, 
+you can use the return statement 
+to synchronously return a value 
+that is placed inside the resulting Future. 
+In other words, you don't have to create a 
+`Future` instance manually 
+inside functions that are marked as `async`. 
+
+The last step is to display the resulting Weather to the user. 
+In Flutter, [`FutureBuilder`]({{site.flutter-api}}/flutter/widgets/FutureBuilder-class.html) and 
+[`StreamBuilder`]({{site.flutter-api}}/flutter/widgets/StreamBuilder-class.html)  
+widgets can be used to display the results of a Future in the UI. 
+In this example, a `FutureBuilder` is used:
+
+<?code-excerpt "lib/async_weather.dart (HomePageWidget)"?>
+```dart
+class HomePage extends StatelessWidget {
+  final HomePageViewModel viewModel = const HomePageViewModel();
+  const HomePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      // Feed a FutureBuilder to your widget tree.
+      child: FutureBuilder<Weather>(
+        // Specify the Future that you want to track.
+        future: viewModel.load(),
+        builder: (context, snapshot) {
+          // A snapshot is of type `AsyncSnapshot` and contains the
+          // state of the Future. By looking if the snapshot contains
+          // an error or if the data is null, you can decide what to
+          // show to the user.
+          if (snapshot.hasData) {
+            return Center(
+              child: Text(
+                snapshot.data.toString(),
+              ),
+            );
+          } else {
+            return const Center(
+              child: CupertinoActivityIndicator(),
+            );
+          }
+        },
+      ),
+    );
+  }
+}
+```
+
+You can see the complete (example on Github)[https://github.com/flutter/website/tree/main/examples/resources/lib/async_weather.dart].
+
+### Leveraging Background Threads or Isolates
+Flutter apps can run on a variety of multi-core hardware, 
+including devices running macOS and iOS. 
+To improve the performance of these applications, 
+it’s sometimes necessary to run tasks 
+on the different cores concurrently- 
+especially to avoid blocking UI rendering 
+on the main thread with long running operations. 
```

Comment:
```suggestion
To improve the performance of these applications, 
you must sometimes run tasks on different cores
concurrently—especially to avoid blocking UI rendering 
on the main thread with long-running operations. 
```